File: C++/build_an_array_with_stack_operations.cpp
// no.1441
#include <vector>
#include <string>
using namespace std;

class Solution {
public:
    vector<string> buildArray(vector<int>& target, int n) {
        vector<string> actions;
        int i = 1;
        int j = 0;
        while(i <= n & j < target.size()){
            actions.push_back("Push");
            if (target[j] != i){
                actions.push_back("Pop");
            }else{
                j++;
            }
            i++;
        }
        return actions;
    }
};

File: python/build_an_array_with_stack_operations.py
# no. 1441
from typing import List

class Solution:
    def buildArray(self, target: List[int], n: int) -> List[str]:
        actions = []
        i, j = 1, 0
        while (i <= n and j < len(target)):
            actions.append('Push')
            if target[j] != i: 
                actions.append('Pop')
            else:
                j += 1
            i += 1
        return actions
                    
